src/tests/system/tests/test_ldap_extra_attrs.py

```diff
@@ -42,3 +48,198 @@ def test_ldap_extra_attrs__filled(client: Client, provider: GenericProvider, att
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_ldap_extra_attrs__check_ldb(client: Client, provider: GenericProvider):
+    """
+    :title: Recently added extra attributes should be in cache db along with their value
+    :setup:
+        1. Create new user "user1"
+        2. Add "firstname:cn, description:gecos, user`s_home_directoy:homeDirectory, user_id:uid, lastname:sn"
+            to ldap_user_extra_attrs
+        3. Start SSSD
+    :steps:
+        1. Run "getent passwd user1" to store user attributes to cache
+        2. Run ldbsearch command
+    :expectedresults:
+        1. User is found
+        2. Result has correct values
+    :customerscenario: True
+    """
+    provider.user("user1").add(gid=111111, uid=100110, gecos="gecos user1", shell="/bin/sh", home="/home/user1")
+    client.sssd.domain[
+        "ldap_user_extra_attrs"
+    ] = "description:gecos, userID:uidNumber, shell:loginShell, groupID:gidNumber"
+    client.sssd.domain["ldap_id_mapping"] = "false"
+    client.sssd.start()
+
+    result = client.tools.getent.passwd("user1")
+    assert result is not None, "getent passwd user1 failed"
+
+    search = client.ldb.search(
+        f"/var/lib/sss/db/cache_{client.sssd.default_domain}.ldb", f"cn=users,cn={client.sssd.default_domain},cn=sysdb"
+    )
+
+    user_dict = search["name=user1@test,cn=users,cn=test,cn=sysdb"]
+    assert user_dict["description"] == ["gecos user1"], "attribute 'description' was not correct"
+    assert user_dict["shell"] == ["/bin/sh"], "attribute 'shell' was not correct"
+    assert user_dict["userID"] == ["100110"], "attribute 'userID' was not correct"
+    assert user_dict["groupID"] == ["111111"], "attribute 'groupID' was not correct"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_ldap_extra_attrs__negative_cache(client: Client, provider: GenericProvider):
+    """
+    :title: When extra attribute of user is add but not assigned, it is neither cached nor displayed
+    :setup:
+        1. Create new user "user1"
+        2. Add "number:telephonenumber" to ldap_user_extra_attrs
+        3. Start SSSD
+    :steps:
+        1. Run "getent passwd user1" to store user to cache
+        2. Run ldbsearch command
+    :expectedresults:
+        1. User is found
+        2. "number" is not in the output
+    :customerscenario: False
+    """
+    provider.user("user1").add()
+
+    client.sssd.domain["ldap_user_extra_attrs"] = "number:telephonenumber"
+
+    client.sssd.start()
+
+    result = client.tools.getent.passwd("user1")
+    assert result is not None, "User is not found"
+    assert result.name == "user1", "User has wrong name"
+
+    search = client.ldb.search(
+        f"/var/lib/sss/db/cache_{client.sssd.default_domain}.ldb", f"cn=users,cn={client.sssd.default_domain},cn=sysdb"
+    )
+
+    user_dict = search["name=user1@test,cn=users,cn=test,cn=sysdb"]
+    with pytest.raises(KeyError):
+        user_dict["number"]
```

Comment:
CodeQL warning here looks valid.

Reply:
Actually it does not. This line is expected to raise ``KeyError`` and if it does not raise, the above line will let the test fail.

Reply:
Based on https://docs.python.org/3/reference/datamodel.html#context-managers

`Typical uses of context managers include saving and restoring various kinds of global state, locking and unlocking resources, closing opened files, etc.`

I don't see the reason for using `with` here(https://peps.python.org/pep-0343/#examples) , but please correct me if I misunderstand.

Reply:
It is because pytest framework kinda abuses the usage of ``with`` statement. See https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions 
`` user_dict["number"]`` is expected to throw an Exception, because ``user_dict`` does not contain pair with "number" as key. This line: ``with pytest.raises(KeyError):`` is catching the exception.

Reply:
Thank you for the explanation